### DISCUSSION TITLE
Best practices for solving Phase-Field equations for a multiphase system

### DISCUSSION THREAD

**Discussion by vermaprk** [idaholab/moose#22780] [Q&A Modules: Phase field] [answered]
I have few questions regarding the use of phase field module available in moose:
1. For solving multiphase problems (eta1, eta2,.....etan) which set of kernels and material property are most efficient? 
2. The summation of etas equal to 1 should be explicitly be constrained or it should come implicitly from solving respective eta PDEs?
3. Can I use ACGrGrMulti instead of AllenCahn kernel for the bulk term in allen cahn equation for solving a multiphase grand potential problem with ACSwitching kernel instead of DerivativeMultiPhaseMaterial material property required earlier by AllenCahn?

Thanks

**Comment by GiudGiud** [comment]
@dschwen @amjokisaari

**Reply by friedmud** [reply]
Pinging @dschwen again...

**Reply by friedmud** [reply]
And maybe @laagesen too

**Comment by dschwen** [comment, marked as answer]
1. I would recommend either the KKS or the Grand Potential formulation. Both are explained in the manual and have examples in the test suite and docs. Side note, we really should write an action to set up both models!! We also have a sub lattice KKS model that we developed and published a year ago, for compound formalism free energies with multiple sublattices.
2. We have a Lagrange multiplier based enforcement. It is not the sum of etas that needs to be one, but the sum of the resulting switching functions, as those determine the phase fractions!
3. The acgrgr stuff is for poly crystal models with a hard coded polynomial free energy. It depends on what you need to model. We also have combined formulations with poly crystal and multi phase support.
